starboard/shared/starboard/player/player_internal.h

```diff
@@ -47,41 +73,36 @@ struct SbPlayerPrivate {
       void* context,
       std::unique_ptr<PlayerWorker::Handler> player_worker_handler);
 
-  void Seek(int64_t seek_to_time, int ticket);
-  template <typename PlayerSampleInfo>
-  void WriteSamples(const PlayerSampleInfo* sample_infos,
-                    int number_of_sample_infos);
-  void WriteEndOfStream(SbMediaType stream_type);
-  void SetBounds(int z_index, int x, int y, int width, int height);
-
-  void GetInfo(SbPlayerInfo* out_player_info);
-  void SetPause(bool pause);
-  void SetPlaybackRate(double playback_rate);
-  void SetVolume(double volume);
-
-  SbDecodeTarget GetCurrentDecodeTarget();
+  void Seek(int64_t seek_to_time, int ticket) override;
+  void WriteSamples(const SbPlayerSampleInfo* sample_infos,
+                    int number_of_sample_infos) override;
+  void WriteEndOfStream(SbMediaType stream_type) override;
+  void SetBounds(int z_index, int x, int y, int width, int height) override;
+  void GetInfo(SbPlayerInfo* out_player_info) override;
+  void SetPause(bool pause) override;
+  void SetPlaybackRate(double playback_rate) override;
+  void SetVolume(double volume) override;
+
+  SbDecodeTarget GetCurrentDecodeTarget() override;
   bool GetAudioConfiguration(
       int index,
-      SbMediaAudioConfiguration* out_audio_configuration);
+      SbMediaAudioConfiguration* out_audio_configuration) override;
 
-  ~SbPlayerPrivate() {
-    --number_of_players_;
-    SB_DLOG(INFO) << "Destroying SbPlayerPrivate. There are "
-                  << number_of_players_ << " players.";
-  }
+  ~SbPlayerPrivateImpl() override;
 
  private:
-  SbPlayerPrivate(SbMediaAudioCodec audio_codec,
-                  SbMediaVideoCodec video_codec,
-                  SbPlayerDeallocateSampleFunc sample_deallocate_func,
-                  SbPlayerDecoderStatusFunc decoder_status_func,
-                  SbPlayerStatusFunc player_status_func,
-                  SbPlayerErrorFunc player_error_func,
-                  void* context,
-                  std::unique_ptr<PlayerWorker::Handler> player_worker_handler);
-
-  SbPlayerPrivate(const SbPlayerPrivate&) = delete;
-  SbPlayerPrivate& operator=(const SbPlayerPrivate&) = delete;
+  SbPlayerPrivateImpl(
+      SbMediaAudioCodec audio_codec,
+      SbMediaVideoCodec video_codec,
+      SbPlayerDeallocateSampleFunc sample_deallocate_func,
+      SbPlayerDecoderStatusFunc decoder_status_func,
+      SbPlayerStatusFunc player_status_func,
+      SbPlayerErrorFunc player_error_func,
+      void* context,
+      std::unique_ptr<PlayerWorker::Handler> player_worker_handler);
+
+  SbPlayerPrivateImpl(const SbPlayerPrivateImpl&) = delete;
```

Comment:
should we move disallow copy and assign to the interface `SbPlayerPrivate` by default?

Reply:
Done.